tests/test_ragas.py

```diff
@@ -0,0 +1,161 @@
+        # SETUP EVERYTHING WE NEED FOR THE TESTS
+        ########################################################################
+
+        # These are the variables which will control the flow of the test.
+        # Since we have to re-construct some Ragas components under the hood,
+
+        student_model_response = "Paris"
+        user_question = "What is the capital of France?"
+        golden_answer = "The capital of France is Paris."
+        base_ds = [{"user_input": user_question, "reference": golden_answer}]
+        mocked_metric = "mocked-metric"
+        mocked_metric_score = 4.0
+
+        # The following section takes care of mocking function return calls.
+        # Ragas is tricky because it has some complex data structures under the hood,
+        # so what we have to do is configure the intermediate outputs that we expect
+        # to receive from Ragas.
+
+        mock_get_metrics.return_value = [mocked_metric]
+        interim_df = DataFrame(
+            {
+                "user_input": [user_question],
+                "response": [student_model_response],
+                "reference": [golden_answer],
+            }
+        )
+        mock_generate_answers_from_model.return_value = interim_df.copy()
+        mocked_evaluation_ds = EvaluationDataset.from_pandas(interim_df)
+        mock_ChatOpenAI.return_value = MagicMock()
+
+        # Ragas requires this value to instantiate an EvaluationResult object, so we must provide it.
+        # It isn't functionally used for our purposes though.
+
+        _unimportant_ragas_traces = {
+            "default": ChainRun(
+                run_id="42",
+                parent_run_id=None,
+                name="root",
+                inputs={"system": "null", "user": "null"},
+                outputs={"assistant": "null"},
+                metadata={"user_id": 1337},
+            )
+        }
+        mock_evaluate.return_value = EvaluationResult(
+            scores=[{mocked_metric: mocked_metric_score}],
+            dataset=mocked_evaluation_ds,
+            ragas_traces=_unimportant_ragas_traces,
+        )
+
+        ########################################################################
+        # Run the tests
+        ########################################################################
+
+        # Configure all other inputs that Ragas does not depend on for proper mocking
+        student_model = ModelConfig(
+            base_url="https://api.openai.com",
+            model_name="pt-3.5-turbo",
+            api_key="test-api-key",
+        )
+        run_config = RunConfig(max_retries=3, max_wait=60, seed=42, timeout=30)
+        evaluator = RagasEvaluator()
+
+        ########################################################################
+        # Test case: directly passing a dataset
+        ########################################################################
+        result = evaluator.run(
```

Comment:
Here too maybe we can call mock gpt-4o with pre-defined evaluation result. This way we only test the evaluation given the student model response, reference and judge model output

Reply:
@abhi1092 It is getting mocked, see: https://github.com/instructlab/eval/pull/193/files#diff-07ba3868aa4f9cd9324d6bfebe02ad4b4a3928285c0f01d29ef3d12d3f3b09b4R94